executor/prepared.go

```diff
@@ -130,16 +130,19 @@ func (e *PrepareExec) Next(ctx context.Context, req *chunk.RecordBatch) error {
 		return ErrPrepareMulti
 	}
 	stmt := stmts[0]
-	if _, ok := stmt.(ast.DDLNode); ok {
-		return ErrPrepareDDL
-	}
 	err = ResetContextOfStmt(e.ctx, stmt)
 	if err != nil {
 		return err
 	}
 	var extractor paramMarkerExtractor
 	stmt.Accept(&extractor)
 
+	// DDL Statements can not accept parameters
+	_, ok := stmt.(ast.DDLNode)
```

Comment:
```
	if _, ok := stmt.(ast.DDLNode); ok && len(extractor.markers) > 0 {
		return ErrPrepareDDL
	}
```

Reply:
is that a Go idiom to limit the scope of `ok`?  In any case I'll make that change.

Reply:
Maybe not a Go idiom, but it is idiom in TiDB when we use type assertion, like:

```
if _, ok := a.Plan.(*plannercore.CheckTable); ok {
		err = ctx.InitTxnWithStartTS(ctx.GetSessionVars().SnapshotTS)
}
```

Reply:
got it.  done.